release-22.1: ui: use plan gist instead of plan id

maryliag opened the pull request:
Backport 1/1 commits from #86653.

/cc @cockroachdb/release

---

Previously, we were using plan id on the table
listing plans, but that was hard to link back to stats.
This commit changes to use gist instead.
It also adds the plan gist on the display of the actual
plan.

Fixes #85272
<img width="1600" alt="Screen Shot 2022-08-23 at 8 27 29 AM" src="https://user-images.githubusercontent.com/1017486/186157925-dad1e120-a1f6-4f08-a150-ce18076c625a.png">
<img width="338" alt="Screen Shot 2022-08-23 at 8 27 35 AM" src="https://user-images.githubusercontent.com/1017486/186157939-f4753ee5-2d16-481d-94ce-daf3e7c076c1.png">
<img width="718" alt="Screen Shot 2022-08-23 at 8 27 44 AM" src="https://user-images.githubusercontent.com/1017486/186157949-fe37f1b8-900c-4091-9c6f-453387d78fad.png">



Release justification: low risk, high benefit change
Release note (ui change): Use plan gist instead of
plan ID on plans table inside the Explain Plan tab of Statement
Details. Also add the plan gist as the first line on the actual
Explain Plan display.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/86872)
<!-- Reviewable:end -->

maryliag committed:
ui: use plan gist instead of plan id

Previously, we were using plan id on the table
listing plans, but that was hard to link back to stats.
This commit changes to use gist instead.
It also adds the plan gist on the display of the actual
plan.

Fixes #85272

Release justification: low risk, high benefit change
Release note (ui change): Use plan gist instead of
plan ID on plans table inside the Explain Plan tab of Statement
Details. Also add the plan gist as the first line on the actual
Explain Plan display.
